### ISSUE TITLE
General solver for SDD matrices

### ISSUE THREAD

**Issue by mengxixi** [danspielman/Laplacians.jl#17]
In the [documentation here](http://danspielman.github.io/Laplacians.jl/v0.1/usingSolvers/index.html#solving-linear-equations-in-laplacians-and-sdd-matrices) it says it's possible to write a wrapper for general SDD matrices that don't necessarily satisfy the M-matrix condition, given that the doc I'm referring to is quite old, I was wondering has this been implemented yet? If not, are there plans for it in the future? Thanks!

**Comment by danspielman** [comment]
That is a good question.  We never implemented one.  If you need that, I could easily write one.  The easy version would be a little inefficient, but it need to be too bad.  

Perhaps the reason I never wrote one is that the right way of doing it depends on the sort of problems to be solved.  Will your systems be positive definite?

**Comment by mengxixi** [comment]
Not necessarily, but if there is an easy and efficient implementation that works only for positive definite systems it would still be great, as we can afford to increase the diagonal slightly to make it PD 
(our systems are positive semidefinite). I guess it could just be a wrapper around the `approxchol_sddm` solver?  This is the one I'm mainly using at the moment.
